### PR TITLE
Feature/tempo

### PR DESCRIPTION
**Pull request by MadsHT**
**Description of your changes:**

Tempo:
  * Added Tempo-distributed umbrella chart
  * Added ckt-claim.yaml to claim a ceph-buket
  * This App has been tested by using Jaegers [Hotrod application](https://github.com/jaegertracing/jaeger/tree/main/examples/hotrod) to send traces Tempo
  
image
image
image

Prometheus:
  * Set `serviceMonitorSelectorNilUsesHelmValues` to false, so Prometheus will read any serviceMonitor resource, before it only read the ones that had the label `release: prometheus`

Global:
  * Added .gitignore to ignore `*.tgz` files and `*.lock` files

Checklist:

* [x] I have bumped the chart version
* [x] I have documented my changes if this is needed
* [x] I have described my changes in the "description of your changes" field
* [x] I have tested this change on a running cluster and the Argocd dashboard shows healthy and synced
* [x] I have created the necessary tests in the chart, if they are possible/necessary
* [x] I have squashed commits if necessary

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+*.lock
+
+charts/**/chart/charts/**.tgz
```
Comment (skovbakke) [inline .gitignore:3]: Could we just do this?
```suggestion
*.tgz
```

Comment (MadsHT) [reply .gitignore:3]: We could yes.

File: charts/prometheus/chart/values.yaml [hunk 2]
```diff
@@ -1,3 +1,3 @@
 kube-prometheus-stack:
   grafana:
-    enabled: false
+    enabled: false
```
Comment (skovbakke) [inline charts/prometheus/chart/values.yaml]: Make sure there is a new-line at the end of the file.

File: charts/tempo-distributed/chart/templates/bkt-claim.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{{- if .Values.bucketClaim.enabled -}}
+apiVersion: objectbucket.io/v1alpha1
+kind: ObjectBucketClaim
+metadata:
+  name: {{ .Values.bucketClaim.name}}
+spec:
+  bucketName: {{ .Values.bucketClaim.bucketName }}
+  storageClassName: {{ .Values.bucketClaim.storageClassName }}
+{{- end -}}
```
Comment (skovbakke) [inline charts/tempo-distributed/chart/templates/bkt-claim.yaml:9]: as above.

File: charts/tempo-distributed/chart/values.yaml [hunk 4]
```diff
@@ -0,0 +1,627 @@
+bucketClaim:
+  enabled: false
+  bucketName: tempo-bucket
+  storageClassName: ceph-bucket
+
+tempo-distributed:
```
Comment (skovbakke) [inline charts/tempo-distributed/chart/values.yaml:6]: We should not include verbatim values from the values file in the the dependency. Please only have the values that we set to something different that the default.

**Review by skovbakke — CHANGES_REQUESTED** [review]

File: charts/tempo-distributed/chart/values.yaml (hunk 4 above)
Comment (MadsHT) [reply charts/tempo-distributed/chart/values.yaml]: I have moved values for tempo out, but the values for the bucket claim should be in the values files since it's a template created by me

Comment (skovbakke) [reply charts/tempo-distributed/chart/values.yaml]: Now there is no values for tempo. Does it work with all default values?

Comment (MadsHT) [reply charts/tempo-distributed/chart/values.yaml]: It should work, it will just not use an s3 bucket fra Ceph, and use a local storage (I guess in memory).
The other values that I removed was setting different ports to deliver traces on, depending on the type it is, Jaeger, zipkin or otlp, and I suspect that it'll be nice to open and close these without having to building a new chart.
Same goes for enabling serviceMonitor and the searchable backend feature.

TL:DR
It should it will just use a local storage instaed of the ceph bucket.

**Review by skovbakke — APPROVED** [review]
LGTM, I have not done any testing though.

File: charts/tempo-distributed/chart/values.yaml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+bucketClaim:
```
[the remark this answers is not in the document]
Comment (MadsHT) [reply charts/tempo-distributed/chart/values.yaml:1]: Alright I will add it.

File: charts/tempo-distributed/chart/values.yaml [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+  # Enableing the bucket claim
+  enabled: false
```
[the remark this answers is not in the document]
Comment (MadsHT) [reply charts/tempo-distributed/chart/values.yaml:3]: The ceph-bucket-claim resource is not deployed.

**Comment by jacobcrawford** [comment]
I guess we also want to install the chart in our GitHub workflow. It can be done by adding the chart to this file: https://github.com/distributed-technologies/helm-charts/blob/main/.github/configs/ct-install.yaml

File: charts/tempo-distributed/chart/values.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (MadsHT) [reply charts/tempo-distributed/chart/values.yaml:1]: Added to the bottom of the file, where the `tempo-distributed: ` obj starts

**Review by skovbakke — APPROVED** [review]
LGTM 😊

**Review by jacobcrawford — APPROVED** [review]
LGTM.
